flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperatorTest.java

```diff
@@ -0,0 +1,150 @@
+public class LegacyTemporalProcessTimeJoinOperatorTest {
+
+	private int keyIdx = 0;
+	private InternalTypeInfo<RowData> rowType = InternalTypeInfo.ofFields(
+			new BigIntType(),
+			new VarCharType(VarCharType.MAX_LENGTH));
+	private TypeInformation<RowData> keyType = InternalTypeInfo.ofFields();
+	private BinaryRowDataKeySelector keySelector = new BinaryRowDataKeySelector(
+			new int[]{keyIdx},
+			rowType.toRowFieldTypes());
+	private InternalTypeInfo<RowData> outputRowType = InternalTypeInfo.ofFields(
+			new BigIntType(),
+			new VarCharType(VarCharType.MAX_LENGTH),
+			new BigIntType(),
+			new VarCharType(VarCharType.MAX_LENGTH));
+	private RowDataHarnessAssertor assertor = new RowDataHarnessAssertor(outputRowType.toRowFieldTypes());
+	private String funcCode =
+			"public class ProcTimeTemporalJoinCondition extends org.apache.flink.api.common.functions.AbstractRichFunction implements org.apache.flink.table.runtime.generated.JoinCondition {\n"
+					+ "\n"
+					+ "    public ProcTimeTemporalJoinCondition(Object[] reference) {\n"
+					+ "    }\n"
+					+ "\n"
+					+ "    @Override\n"
+					+ "    public boolean apply(org.apache.flink.table.data.RowData in1, org.apache.flink.table.data.RowData in2) {\n"
+					+ "        return true;\n"
+					+ "    }\n"
+					+ "}\n";
+
+	private GeneratedJoinCondition joinCondition = new GeneratedJoinCondition(
+			"ProcTimeTemporalJoinCondition",
+			funcCode,
+			new Object[0]);
+
+	/** test proctime temporal join when non state clear. **/
+	@Test
+	public void testProcTimeTemporalJoin() throws Exception {
+		LegacyTemporalProcessTimeJoinOperator joinOperator = new LegacyTemporalProcessTimeJoinOperator(
+				rowType,
+				joinCondition,
+				0,
+				0);
+		KeyedTwoInputStreamOperatorTestHarness<RowData, RowData, RowData, RowData> testHarness = createTestHarness(
+				joinOperator);
+		testHarness.open();
+		testHarness.setProcessingTime(1);
+		testHarness.processElement1(insertRecord(1L, "1a1"));
+
+		testHarness.setProcessingTime(2);
+		testHarness.processElement2(insertRecord(2L, "2a2"));
+
+		testHarness.setProcessingTime(3);
+		testHarness.processElement1(insertRecord(2L, "2a3"));
+
+		testHarness.setProcessingTime(4);
+		testHarness.processElement2(insertRecord(1L, "1a4"));
+
+		testHarness.setProcessingTime(5);
+		testHarness.processElement1(insertRecord(1L, "1a5"));
+
+		List<Object> expectedOutput = new ArrayList<>();
+		expectedOutput.add(insertRecord(2L, "2a3", 2L, "2a2"));
+		expectedOutput.add(insertRecord(1L, "1a5", 1L, "1a4"));
+		assertor.assertOutputEquals("output wrong.", expectedOutput, testHarness.getOutput());
+		testHarness.close();
+	}
+
+	/** test proctime temporal join when set state clear. **/
+	@Test
+	public void testStateCleanUp() throws Exception {
+		final int minRetentionTime = 10;
+		LegacyTemporalProcessTimeJoinOperator joinOperator = new LegacyTemporalProcessTimeJoinOperator(
+				rowType,
+				joinCondition,
+				minRetentionTime,
+				minRetentionTime * 3 / 2);
```

Comment:
`final int maxRententionTime = minRetentionTime * 3 / 2`
we can reuse  `maxRententionTime` in following code

Reply:
fixed.